### ISSUE TITLE
Change Request: Return successfully passing rules count from lint methods of Node API

### ISSUE THREAD

**Issue by brettz9** [eslint/eslint#18013]
### ESLint version

8.56.0

### What problem do you want to solve?

Hi,

I have a formatter (at https://github.com/brettz9/eslint-formatter-badger ) which intends to print a badge indicating the number of rules failing out of how many rules are applied, but it is currently unable to get the latter (i.e., the total number of rules in effect).

The results might look like:

![cli-no-failures](https://github.com/eslint/eslint/assets/20234/5af9c994-473a-470f-a019-a7da7a2cf80e)

or:

![cli-1-failing-suggestion-medium-percentage-threshold-plural](https://github.com/eslint/eslint/assets/20234/2d6261fd-6cf7-4449-a4c1-ddcded2407b0)

(While it may not seem so important to have a badge to communicate linting results compared to say testing results, it does convey a sense of how much consideration went into checking the best practices (including for say security) of the code.)

The `getRulesMetaForResults` method results only indicate the failing tests, and `builtinRules` only indicates the total rules available, not the total rules in effect.

### What do you think is the correct solution?

I would like a way to reflect on the successfully passing rules supplied to the ESLint constructor (e.g., with `overrideConfigFile`) so that the badge can display how many rules are passing as well as how many may be failing.

I would think the value (say `totalApplicableRules`) could be returned along with other `LintResult` values such as `fixableErrorCount` returned from the likes of `ESLint#lintFiles` and `ESLint#lintText`.

### Participation

- [X] I am willing to submit a pull request for this change.

### Additional comments

I think I should be able to the PR myself, but no guarantees.

**Comment by nzakas** [comment]
> I would like a way to reflect on the successfully passing rules supplied to the ESLint constructor (e.g., with `overrideConfigFile`) so that the badge can display how many rules are passing as well as how many may be failing.

Can you explain what you mean by "reflect"? It seems like the request is just for a count of rules that were run, but I'm not 100% sure based on this sentence.

**Comment by nzakas** [comment]
And you're looking for all the rules that were executed in a single run, across all files? (As opposed to all of the rules run on a single file?)

**Comment by brettz9** [comment]
Right, all the rules that were executed in a single run, across all files.

**Comment by nzakas** [comment]
Got it. This seems like it would be easy to add, but would like @eslint/eslint-team to weigh in with other opinions before marking as accepted.

**Comment by snitin315** [comment]
I'm fine with this enhancement.

**Comment by nzakas** [comment]
@snitin315 famous last words.  :)

**Comment by Rec0iL99** [comment]
ping @eslint/eslint-team

**Comment by JoshuaKGoldberg** [comment]
👍 I'm in favor

**Comment by mdjermanovic** [comment]
I think we can consider adding this. However, this doesn't look like a trivial change, so I think it needs an [RFC](https://github.com/eslint/rfcs).

**Comment by coderaiser** [comment]
If I understand correctly it is already possible with help o Node API methods:
- [`eslint.calculateConfigForFile()`](https://eslint.org/docs/latest/integrate/nodejs-api#-eslintcalculateconfigforfilefilepath);
- [`eslint.lintText()`](https://eslint.org/docs/latest/integrate/nodejs-api#-eslintlinttextcode-options);

The only thing to remember is count of passed rules can vary according to overrides applied to file groups so it can be different for each file.

Here is possible code example:

```js
const {readFile} = require('fs/promises');
const {ESLint} = require('eslint');
const eslint = new ESLint();
const {keys} = Object;

const files = ['/index.js', '/client.js'];

let allExecutedCount = 0;
let sallScceededRules = 0;

for (const file of files) {
    const code = await readFile(file);
    const config = await eslint.calculateConfigForFile(file);
    const rules = keys(config.rules);
    const executedRules = new Set(rules);  
    const succeededRules = new Set(rules);

    const results = await eslint.lintText(file, config);
    
    for (const {messages} of results) {
        for (const {ruleId} of messages) {
            succeededRules.delete(ruleId);
        }
    }
    
    console.log(`Executed Rules Count for file '${file}': ${executedRules.size}`);
    console.log(`Succeeded Rules Count for file '${file}': ${successededRules.size}`);

    if (allExecutedCount < executedRules.size)
        allExecutedCount = executedRules.size;
}

console.log(`All executed rules count: ${allExecutedCount}`);
```

**Comment by nzakas** [comment]
@coderaiser yes, it's possible to get that information if you're using the API directly. In this case, the request is to expose that information inside of formatters.
